tests/func/test_full.py

```diff
@@ -37,3 +37,21 @@ def test_run(psql):
     WHERE description = 'mail: alice@ldap2pg.docker';
     """)
     assert comment
+
+
+def test_role_config(psql):
+    # type: (PSQL) -> None
+
+    from sh import ldap2pg
+    c = 'tests/func/ldap2pg.config.yml'
+
+    # Synchronize all
+    ldap2pg('-N', c=c)
+
+    roles = list(psql.roles())
+
+    assert 'config_test_role' in roles
+    assert {
+               'log_statement': 'ddl',
+               'application_name': 'config_test_application_name',
+    } == psql.config('config_test_role')
```

Comment:
Cool.

I'd prefer more atomic, single line, asserts. But that may be juste a matter of taste.

Could you add some `ALTER ROLE SET TO` in `fixtures/postgres.sh` and test here the reset of such config ?

Reply:
Sure thing. I expanded the function test to cover:
- create a new role
- modify an existing role
- reset permissions on an existing role
- leave an existing role alone

I think the last one addresses your regression comment above.  I changed the behavior so that this new feature is strictly "opt-in".  If users don't specify the `config` block in their yaml file then we won't make any effor to sync the configs. If someone wants to reset all of the role configs back to default then they can explicitly do that with a `config: {}` in the yaml file.

Let me know if that works for you.

Also, I changed my `assert` style as requested.

As for running the functional tests, I would have expected them to run with no other set up than just typing `docker-compose up` or `make -C tests/func/ tests` or something and to have the makefile/docker-compose/whatever be able to bootstrap the functional test environment from scratch.  I ended up having to run the `ldap2pg/docker-compose.yml` containers manually, and then hack a bunch of the `dsn` strings to connect to `-h localhost`.  This works well enough to run my tests, but I wouldn't want to commit those change.

Reply:
Hi @randvoorhies ,

The opt-in behaviour is good. Let's go this way.

For the development environment. The thing is that I don't want docker-compose.yml to bind 5432. Thus I use dnsdock to resolve container IP.

You can bind localhost:5432 and localhost:389 by adding a `docker-compose.override.yml` file as documented in [hacking](https://github.com/dalibo/ldap2pg/blob/master/docs/hacking.md#docker-development-environment).

Did you have to add more tweaks than documented in docs/hacking.md ?

Another solution would be to have a container dedicated to run ldap2pg instead of running in a Python virtualenv in host. Again, a docker-compose.override.yml should do the trick.

Once you have psql and ldaputils configured, you can use ldap2pg, pytest and any other tools to work with PostgreSQL, OpenLDAP, directly without wrapper. I find this very valuable.